minoc/src/main/java/org/opencadc/minoc/PostAction.java

```diff
@@ -0,0 +1,133 @@
+*  the GNU Affero General Public        la “GNU Affero General Public
+*  License as published by the          License” telle que publiée
+*  Free Software Foundation,            par la Free Software Foundation
+*  either version 3 of the              : soit la version 3 de cette
+*  License, or (at your option)         licence, soit (à votre gré)
+*  any later version.                   toute version ultérieure.
+*
+*  OpenCADC is distributed in the       OpenCADC est distribué
+*  hope that it will be useful,         dans l’espoir qu’il vous
+*  but WITHOUT ANY WARRANTY;            sera utile, mais SANS AUCUNE
+*  without even the implied             GARANTIE : sans même la garantie
+*  warranty of MERCHANTABILITY          implicite de COMMERCIALISABILITÉ
+*  or FITNESS FOR A PARTICULAR          ni d’ADÉQUATION À UN OBJECTIF
+*  PURPOSE.  See the GNU Affero         PARTICULIER. Consultez la Licence
+*  General Public License for           Générale Publique GNU Affero
+*  more details.                        pour plus de détails.
+*
+*  You should have received             Vous devriez avoir reçu une
+*  a copy of the GNU Affero             copie de la Licence Générale
+*  General Public License along         Publique GNU Affero avec
+*  with OpenCADC.  If not, see          OpenCADC ; si ce n’est
+*  <http://www.gnu.org/licenses/>.      pas le cas, consultez :
+*                                       <http://www.gnu.org/licenses/>.
+*
+************************************************************************
+*/
+
+package org.opencadc.minoc;
+
+import java.net.URI;
+
+import org.apache.log4j.Logger;
+import org.opencadc.inventory.Artifact;
+import org.opencadc.inventory.db.ArtifactDAO;
+import org.opencadc.minoc.ArtifactUtil.HttpMethod;
+
+/**
+ * Interface with storage and inventory to update the metadata of an artifact.
+ *
+ * @author majorb
+ */
+public class PostAction extends ArtifactAction {
+    
+    private static final Logger log = Logger.getLogger(PostAction.class);
+
+    /**
+     * Default, no-arg constructor.
+     */
+    public PostAction() {
+        super(HttpMethod.POST);
+    }
+
+    /**
+     * Update artifact metadata.
+     * @param artifactURI The identifier for the artifact. 
+     * @return The artifact
+     */
+    @Override
+    public Artifact execute(URI artifactURI) throws Exception {
+        
+        String newURI = syncInput.getParameter("uri");
+        String newContentType = syncInput.getParameter("contentType");
+        String newContentEncoding = syncInput.getParameter("contentEncoding");
+        log.debug("new uri: " + newURI);
+        log.debug("new contentType: " + newContentType);
+        log.debug("new contentEncoding: " + newContentEncoding);
+        
+        ArtifactDAO dao = getArtifactDAO();
+        Artifact artifact = getArtifact(artifactURI, dao);
+        
+        // TODO: enable modifying URIs when supported by DAO
+        // TODO: how to support clearing values?
```

Comment:
You don't need to make comp0lex logic here to see if something changed. The ArtifactDAO recomputes accMetaChecksum and astModified and only writes to the database if there are changes already.

Not sure about clearing values... only applicable to contentEncoding